src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1694,15 +1696,30 @@ def get_disk_driver(self) -> Optional[ManagedClusterStorageProfileDiskCSIDriver]
                 "--disable-disk-driver at the same time."
             )
 
+        if disable_disk_driver and azuredisk_csi_version:
+            raise ArgumentUsageError(
+                "The parameter --disable-disk-driver cannot be used "
+                "when --azuredisk-csi-version is specified.")
+
+        # Setting default driver as v1 if not mentioned
+        if not azuredisk_csi_version:
+            azuredisk_csi_version = "v1"
+
+        if not (azuredisk_csi_version == "v1" or azuredisk_csi_version == "v2"):
+            raise InvalidArgumentValueError(
+                "Valid values for --azuredisk-csi-version are v1 and v2")
+
         if self.decorator_mode == DecoratorMode.CREATE:
             if disable_disk_driver:
                 profile.enabled = False
             else:
                 profile.enabled = True
+                profile.version = azuredisk_csi_version
```

Comment:
Actually, the code never gets here. There's no enable option for `aks create`.

Reply:
For ```aks create``` , if disable option is false we enter the enabling section.

Reply:
Sorry, didn't notice the condition of return None changed.

Then in an update scenario, if the disk csi version was `v2` and enable option was incorrectly specified. This would set the disk csi version to `v1`?

Reply:
That is a very good point. I need to confirm the intended behavior of the ```aks update``` once. Thanks for pointing this out.

Reply:
We decided to handle the scenario in the RP side. We will not downgrade the driver version while upgradation.